### PR TITLE
Make WaitSet static in rmw_wait

### PR DESCRIPTION
**Pull request by gerkey**
This change is intended to prevent a race that can occur between (the exit of) `rmw_wait()` here and the invocation of `rmw_trigger_guard_condition` at https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/executor.cpp#L179.  The theory is that triggering the guard condition causes it to iterate over its internal list (in our case, of length 1) of wait sets to which it is attached, and that this can happen at exactly the time that one of those wait sets is being destroyed in another thread.

To avoid that race, we never allow the wait set to be destroyed, and instead do our own cleanup (detaching conditions) of its contents on exit from `rmw_wait`.

This change is not meant to be permanent.  Rather, it is a pragmatic change that doesn't sacrifice any functionality (we're already mutually excluding calls to `rmw_wait`) but improves behavior until we add the new features regarding management of wait sets that is being worked on in #96. 

This change appears to fix the segfault mentioned in https://github.com/ros2/system_tests/pull/72#issuecomment-157103620.

### CODE REVIEW

File: rmw_opensplice_cpp/src/rmw_wait.cpp [hunk 1]
```diff
@@ -47,7 +47,27 @@ rmw_wait(
   rmw_clients_t * clients,
   rmw_time_t * wait_timeout)
 {
-  DDS::WaitSet waitset;
+  // We're making this object static to avoid a race between its destructor
+  // being called on exit from this function and an interrupt guard condition
+  // that holds a pointer to it being called in parallel from somewhere else.
+  // This structure will be changed when we add to rmw the feature of supporting
+  // multiple wait sets.
+  static DDS::WaitSet waitset;
```
Comment (dirk-thomas) [inline rmw_opensplice_cpp/src/rmw_wait.cpp]: Should we add a `TODO(...) ...` here?

**Comment by gerkey** [comment]
CI jobs:

* http://ci.ros2.org/job/ci_osx/530/
* http://ci.ros2.org/job/ci_linux/648/
* http://ci.ros2.org/job/ci_windows_connext_static/19/

Other Windows jobs are still in the pipeline.

File: rmw_opensplice_cpp/src/rmw_wait.cpp (hunk 1 above)
Comment (gerkey) [inline rmw_opensplice_cpp/src/rmw_wait.cpp]: 95d99b4

**Comment by dirk-thomas** [comment]
lgtm

**Comment by gerkey** [comment]
Here's the opensplice Windows jobs (the connext ones don't matter for this change):

http://ci.ros2.org/job/ci_windows_opensplice/718/

**Pull request merged**